openstack-poppy/src/main/java/org/jclouds/openstack/poppy/v1/mapbinders/JSONPatchUpdate.java

```
@@ -0,0 +1,84 @@
+import org.jclouds.openstack.poppy.v1.domain.Service;
+import org.jclouds.rest.MapBinder;
+import org.jclouds.rest.binders.BindToJsonPayload;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.github.fge.jsonpatch.diff.JsonDiff;
+import com.google.inject.Guice;
+import com.google.inject.Inject;
+
+/**
+ * This will create a JSONPatch out of a Service and an UpdateService.
+ *
+ * User side:
+ * Get a Service with api.get(service_id)
+ * Get a UpdateService builder by using Service.toUpdatabaleService()
+ *    This step will provide an interface that exposes the updatable JSON values to the user.
+ * Use the UpdateService.Builder instance to modify and build() a new UpdateService.
+ * Send the original Service and the new UpdateService to the api.update method.
+ *
+ * jclouds side:
+ * Convert the Service to UpdateService, but don't change it (this is the source).
+ * Serialize both source and target to String
+ * Diff to create JSONPatch using dependency.
+ * Send the JSONPatch in the request.
+ *
+ * JSONPatch RFC:
+ * https://tools.ietf.org/html/rfc6902
+ */
+public class JSONPatchUpdate implements MapBinder {
+   @Inject
+   private BindToJsonPayload jsonBinder;
+
+   @Override
+   public <R extends HttpRequest> R bindToRequest(R request, Map<String, Object> postParams) {
+      String jsonPatch = null;
+      Service service = (Service) postParams.get("service");
+
+      Json json = Guice.createInjector(new GsonModule()).getInstance(Json.class);
+
+      String targetService = json.toJson(postParams.get("updateService"));
+      String sourceService = json.toJson(service.toUpdatableService().build());
+
+      ObjectMapper mapper = new ObjectMapper();
+      try {
+         jsonPatch = JsonDiff.asJson(mapper.readTree(sourceService), mapper.readTree(targetService)).toString();
+      } catch (IOException e) {
+         e.printStackTrace();
+         throw new RuntimeException("Could not create a JSONPatch");

```

Don't mask the original exception. Use `RuntimeException(String message, Throwable cause)` instead.